programs/drift/src/state/oracle.rs

```diff
@@ -169,6 +172,66 @@ impl OracleSource {
     }
 }
 
+#[derive(Default, Clone, Copy, Debug)]
+pub struct MMOraclePriceData {
+    pub mm_oracle_price: i64,
+    pub mm_oracle_delay: i64,
+    pub oracle_price_data: OraclePriceData,
+}
+
+impl MMOraclePriceData {
+    pub fn default_usd() -> Self {
+        MMOraclePriceData {
+            mm_oracle_price: PRICE_PRECISION_I64,
+            mm_oracle_delay: 0,
+            oracle_price_data: OraclePriceData::default_usd(),
+        }
+    }
+
+    pub fn get_oracle_price(&self) -> i64 {
+        if self.mm_oracle_delay <= self.oracle_price_data.delay && self.mm_oracle_price != 0i64 {
+            self.mm_oracle_price
+        } else {
+            self.oracle_price_data.price
+        }
+    }
+
+    pub fn get_delay(&self) -> i64 {
```

Comment:
should this and confidence be same functions so you choose consistent/correct value for both?